### PR TITLE
NULL to nullptr

### PR DESCRIPTION
**Pull request by SyrianBallaS**
Basically all this does is replace NULL with nullptr, for params that aren't pointers they were replaced with rvalues. Small PRs for now. The little things can sometimes lead to unexpected results.

### CODE REVIEW

File: src/RA_Core.cpp [hunk 1]
```diff
@@ -525,18 +525,18 @@ API BOOL CCONV _RA_OfferNewRAUpdate(const char* sNewVer)
...
 		//	SW_SHOWNORMAL ); 
 
-		ShellExecute(NULL,
+		ShellExecute(nullptr,
 			TEXT("open"),
 			TEXT("http://www.retroachievements.org/download.php"),
-			NULL,
-			NULL,
+			nullptr,
+			nullptr,
```
Comment (Jamiras) [inline src/RA_Core.cpp]: Is it safe to use `nullptr` in the WINAPI calls?

File: src/RA_Core.cpp [hunk 2]
```diff
@@ -752,32 +752,33 @@ API HMENU CCONV _RA_CreatePopupMenu()
 	HMENU hRA_LB = CreatePopupMenu();
 	if (RAUsers::LocalUser().IsLoggedIn())
 	{
+		// TODO: Use the literals once PR 23 is accepted
 		AppendMenu(hRA, MF_STRING, IDM_RA_FILES_LOGOUT, TEXT("Log&out"));
-		AppendMenu(hRA, MF_SEPARATOR, NULL, NULL);
+		AppendMenu(hRA, MF_SEPARATOR, UINT_PTR{}, nullptr);
```
Comment (Jamiras) [inline src/RA_Core.cpp]: I think it would be more standard/common to use `0` instead of `UINT_PTR{}`.

File: src/RA_Defs.h [hunk 3]
```diff
@@ -11,6 +11,8 @@
 #include <queue>
 #include <deque>
 #include <map>
+#include <atlbase.h>
```
Comment (Jamiras) [inline src/RA_Defs.h]: why was this necessary?

File: src/RA_Core.cpp (hunk 1 above)
Comment (SyrianBallaS) [reply src/RA_Core.cpp]: Yes, "far* LP" or "near* P" stuff are pointers. I did notice some NULLs in the AppendMenu were actually supposed to be integers. Windows NT is compatible with C++ if you know how to use it, Microsoft themselves recommend using nullptr.

File: src/RA_Core.cpp (hunk 2 above)
Comment (SyrianBallaS) [reply src/RA_Core.cpp]: Well I was going to use my literals in a different pull request but felt that I shouldn't copy and paste them. <code>UINT_PTR</code> is basically the same as <code>std::size_t</code>. Starting with C++11, rvalue implicit conversions are flagged.

But I know what you mean.

File: src/RA_Defs.h (hunk 3 above)
Comment (SyrianBallaS) [reply src/RA_Defs.h]: It's not. At least it wasn't here. It is needed for CComPtr though. I think I was going to that first but thought I should do something simpler.

**Comment by SyrianBallaS** [comment]
@Jamiras Thanks for the review

File: src/RA_Core.cpp (hunk 2 above)
Comment (SyrianBallaS) [reply src/RA_Core.cpp]: I was looking into it further and <code>UINT_PTR</code> is a variable/polymorphic type, it's either <code>uintptr_t</code>, <code>unsigned int</code>, or <code>unsigned __int64</code> depending on where you got it from. So I believe it's more appropriate to hold off on it for correctness. The other standard way to do it would be <code>UINT_PTR{0}</code> but that's pretty much the same thing. <code>type{}</code> just means <code>default</code>.

File: src/RA_Defs.h (hunk 3 above)
Comment (SyrianBallaS) [reply src/RA_Defs.h]: This is gone
